fuel stop advisor: Add service for data simulation

blochm opened the pull request:
In order to have data simulated in Fuel Stop Advisor, need to add
automotive message broker with a particular configuration file in
dbus session and systemd user.

[GDP-15] Part of the Integrate LBS / FSA into GDP

blochm committed:
fuel stop advisor: Add service for data simulation

In order to have data simulated in Fuel Stop Advisor, need to use
automotive message broker with a configuration file for data
simultation injection in dbus session and systemd user.

[GDP-15] Part of the Integrate LBS / FSA into GDP

zeenix commented on meta-genivi-dev/recipes-extended/automotive-message-broker/automotive-message-broker/ambd_fsa.service:
```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=Automotive Message Broker for FSA configuration file
```
The description is supposed to be of the service/unit, not the file.

zeenix commented on meta-genivi-dev/recipes-extended/automotive-message-broker/automotive-message-broker_git.bb:
```diff
@@ -31,8 +32,13 @@ do_install_append() {
     install -m 0644 ${WORKDIR}/ambd.service ${D}${systemd_unitdir}/system
+
+    install -d ${D}${libdir}/systemd/user/
+    install -m 0644 ${WORKDIR}/ambd_fsa.service ${D}${libdir}/systemd/user
```
Why does this service need to run as part of user instance while AMBD itself runs on system instance?

blochm replied:
The process that look for the interface is on the dbus session, after all the other process are in dbus session.

zeenix commented on meta-genivi-dev/recipes-extended/automotive-message-broker/automotive-message-broker_git.bb:
```diff
@@ -17,6 +17,7 @@ SRCREV = "c216955d16ca275159891cad296217094d972390"
 
 SRC_URI += "file://amb_allow_sessionbus.patch \
             file://ambd.service \
+            file://ambd_fsa.service \
```
Is there a reason we want this FSA-specific service installed by AMB recipe rather than an FSA-specific recipe?

blochm replied:
In fact, it is a amb instance with a particular config file that is used to simulate data. It is used for FSA but it is an amb releated.
I see like dbus: 
    - There is a service for dbus session
    - There is a service for dbus server
but at the end the process is still dbus-dameon with parameters specific.

zeenix replied:
Thanks. I see your point but I don't agree. You're launching AMD instance but you're launching it for FSA with FSA-specific configuration (correct?) so that just means FSA has a dependency on this recipe.

zeenix replied:
Hmm.. actually is this a correct usage of AMB? Are you supposed to launch multiple instances of it on different buses?

blochm replied:
Some options:
For sure, I need to use automobile message borker.
Then, 
1. I use the one in the sytem dbus WHERE I need to add my configuration file (where someone that use amb may brake something fro him)
2. Or I use the one in the session dbus with my configuration file in the session where no one use it.

zeenix replied:
You put services on session bus to keep them private to each user, not to keep them private from other apps. I checked through the only AMB docs I could find with a quick searching but I didn't find any mention of the use case of having to launch your own AMB demon with different configuration but only that you might need plugins for additional use cases (e.g getting GPS support). Is that what you're doing? If you could point to relevant AMB docs, that would help a lot.

In the meantime, I've asked @gunnarx to also review here since I believe he knows a bit more about AMB.

zeenix reviewed (changes requested):
* The prefix in shortlog is name of the module/component being changed.

* Good work on the log actually but it's still a bit vague: "particulr configuration file" -> best describe what file you mean. I feel the issue is mainly that this should not be done as part of ambd recipe but the one that adds the FSA data simulation.

blochm reviewed (commented):
- Changes the service file description.
- Add comments on questions.

zeenix reviewed (commented):
Please squash the fixing commit to the original commit. Overwritting history is the only way you can fix commit log.

blochm commented:
Ok Done for the squash

gunnarx commented:
@blochm Where is that particular configuration file that you use for ambd?
I'd like to evaluate if there is a risk of breaking anything else.  If there isn't we just add it to the main one.

Because not much else using AMB to my knowledge, and we don't expect it in the future since it should be replaced by a new VSS/VSI service (unless there is some reasonable combination to make with AMB+VSS/VSI)

There are already simulators under way in order to support W3C VISS spec which is based on VSS.
I imagine the GDP future to be that either your log service could be replaced by such a W3C compatible simulator, or your log service would be rewritten on top of VSS/VSI.  
But that's for later - let's see how to best get this running now.

gunnarx commented:
Ref related PR: https://github.com/GENIVI/hmi-layout-gdp/pull/10 Add FSA to main menu.

asanoaozora commented:
Reason why we patched the AMB to run it under the session DBus is mainly for convenience. Please consider the fact that I mainly develop and extend the FSA under Ubuntu. Matthias is in charge of Yocto recipe, GDP build and testing (that's is very helpful :-)).
By using this patch, it's more easy for me to test. 
And when another solution will be mature (with the same features, especially the plugin mechanism, that is convenient) I'll replace AMB.

zeenix commented:
@asanoaozora Thanks for explaining but developer's ease alone is not a valid argument for decision of which bus is to be used. If it's a system bus (which it is, at least in case of GDP and automotive in general), it should be running on system bus. Session bus is only for per-login/session services.

blochm committed:
fuel stop advisor: Positioning: Update Version and Add service

In order to have all FSA software aligned, version has been
updated. To launch enhanced-position-service automatically,
a service file was created.

[GDP-15] Part of the Integrate LBS / FSA into GDP

blochm committed:
fuel stop advisor: Navit: Update Version and bug fixing

In order to have all FSA software aligned, version has been
updated. Also, some bug fixing has be done due to this
version update and changes where an include should be put.

[GDP-15]  Part of the Integrate LBS / FSA into GDP
[GDP-562] Part of Navigation build sometimes breaks

blochm committed:
fuel stop advisor: Navigation-service: Update Version and Add services

In order to have all FSA software aligned, version has been
updated. It also fix where to find include files used in navit plugins.
To launch navit plugins that belongs to navigation-service and poi-server
automatically, service files were created.

[GDP-15] Part of the Integrate LBS / FSA into GDP
[GDP-562] Part of Navigation build sometimes breaks

blochm committed:
fuel stop advisor: Navigation-application: Update Version and Add services

In order to have all FSA software aligned, version has been updated.
To launch fuel-stop-advisor and hmi-launcher proceses automatically,
service files were created.

[GDP-15] Part of the Integrate LBS / FSA into GDP

blochm committed:
navit: Include again Navit & all it's dependents
Merge master into wip/cool-new-fsa

blochm committed:
Remove issue with defined but not used

blochm committed:
Bugfix on opengl -> opengles2

gunnarx commented:
1. There is no further development on this m-g-d repo since m-g-d is now included in g-d-p repo.

2. I think this PR is also superseded by https://github.com/GENIVI/genivi-dev-platform/pull/150 which has the same (possibly updated) content from what I can see.

For those reasons, closing the PR.
